adr/component_structure.md

```diff
@@ -0,0 +1,102 @@
+# Create both low-level and high-level components that are exported from each package.
+
+## Context
+
+The primary issue we are facing right now is determining how "high-level" the component should be. For example, should we create a single component with many props, or should we create many components to be composed together.
+
+```jsx
+<ClayCard>
+	<ClayCard.Title onClick={handleTitleClick}>
+		<ClayIcon symbol="check" />
+
+		{'Hello World'}
+	</ClayCard.Title>
+
+	<ClayCard.Body onClick={handleBodyClick}>
+		<ClayCard.Description>
+			{'Some cool descriptors'}
+		</ClayCard.Description>
+	</ClayCard.Body>
+</ClayCard>
+
+// vs
+<ClayCard
+	iconTitleLeftSymbol="check"
+	onTitleClick={handleTitleClick}
+	onBodyClick={handleBodyClick}
+	description="foo bar baz"
+	title="hello world"
+/>
+```
+
+## Decision
+
+The decision going forward is to create 2 sets of components
+
+-   low-level composable components such as `<ClayCard.Header>`, `<ClayCard.Footer>`
+-   simple high-level components that cover basic lexicon uses
+
+### What does this look like?
+
+#### Composable:
+
+![Composable Drop Down](./images/component_structure_2.png)
+
+```jsx
+<ClayDropDown>
+	<ClayDropDown.Search
+		onChange={() => {}}
+		placeholder="Search"
+		spritemap={spritemap}
+		value={''}
+	/>
+
+	<ClayDropDown.ItemList>
+		{items.map((item, i) => (
+			<ClayDropDown.Item href={item.href} key={i}>
+				{item.label}
+			</ClayDropDown.Item>
+		))}
+	</ClayDropDown.ItemList>
+</ClayDropDown>
+```
```

Comment:
Similarly to the example below. Wouldn't it be interesting to show how we think this exact same component feels like on high-level? That way we can compare and maybe come up with clearer guidelines on what would work and what wouldn't

Reply:
I probably wouldn't include that here because it'd be too difficult to condense into an example without adding all of the business logic. Plus this example isn't to prove a point that composable is better, its just to point out that you are able to compose components together.

Reply:
Ok, as long as that's clear I'm fine with it 👍